src/psyclone/tests/psyad/tl2ad_test.py

```diff
@@ -403,9 +405,10 @@ def test_generate_adjoint_kind(fortran_reader, fortran_writer):
     assert ad_fortran_str in expected_ad_fortran_str
 
 
-def test_generate_adjoint_multi_kernel(fortran_reader):
-    '''Check that generate_adjoint raises the expected error when there
-    are multiple kernels in a module.
+def test_generate_adjoint_multi_kernel(fortran_reader, fortran_writer):
```

Comment:
If poss, please could you extend this to do compilation too?

Reply:
OK. Added.